src/components/Navigation/PageMenu/PageMenu.test.tsx

```diff
@@ -109,3 +109,236 @@ test('Hides children when onlyShowSelectedChildren = true', () => {
+      ),
+    }
+  );
+
+  const user = userEvent.setup();
+
+  for (const item of items) {
+    expect(screen.queryAllByText(item.label).length).toBe(2);
+  }
+
+  const button = screen.getByRole('button', { name: items[0].label });
+
+  expect(button).toBeDisabled();
+  await user.click(button);
+
+  const section = document.querySelector(`#${items[0].value}`)!;
+
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  expect(section.scrollIntoView).not.toHaveBeenCalled();
+
+  // Expect other button _not_ to be disabled
+  const otherButton = screen.getByRole('button', { name: items[1].label });
+
+  expect(otherButton).not.toBeDisabled();
+
+  await user.click(otherButton);
+
+  const otherSection = document.querySelector(`#${items[1].value}`)!;
+
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  expect(otherSection.scrollIntoView).toHaveBeenCalled();
+});
+
+test('OnClick runs as expected with border variant', async () => {
+  const items = fakeItems();
+
+  render(
+    <div>
+      <PageMenu variant="border" />
+      {items.map((item) => (
+        <Section key={item.value} label={item.label} value={item.value} />
+      ))}
+    </div>,
+    {
+      wrapper: (props: { children: ReactNode }) => (
+        <PageMenuProvider items={items}> {props.children}</PageMenuProvider>
+      ),
+    }
+  );
+
+  const user = userEvent.setup();
+
+  for (const item of items) {
+    expect(screen.queryAllByText(item.label).length).toBe(2);
+  }
+
+  const button = screen.getByRole('button', { name: items[1].label });
+
+  await user.click(button);
+
+  const section = document.querySelector(`#${items[1].value}`)!;
+
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  expect(section.scrollIntoView).toHaveBeenCalled();
+});
+
+test('Hides children when onlyShowSelectedChildren = true', () => {
+  const items = fakeItems(true);
+
+  render(
+    <div>
+      <PageMenu onlyShowSelectedChildren variant="border" />
+      {items.map((item) => (
+        <Section key={item.value} {...item}>
+          {item.children?.map((child) => (
+            <h2 key={child.value} id={child.value}>
+              {child.label + 'content'}
+            </h2>
+          ))}
+        </Section>
+      ))}
+    </div>,
+    {
+      wrapper: (props: { children: ReactNode }) => (
+        <PageMenuProvider items={items}>{props.children}</PageMenuProvider>
+      ),
+    }
+  );
+
+  for (const child of items[0].children ?? []) {
+    expect(screen.getByText(child.label)).toBeInTheDocument();
+  }
+
+  for (const item of items.slice(1)) {
+    for (const child of item.children ?? []) {
+      expect(screen.queryByText(child.label)).not.toBeInTheDocument();
+    }
+  }
+});
+
+test('Does not run OnClick when disabled', async () => {
+  const items = fakeItems().map((item, index) => ({
+    ...item,
+    disabled: index % 2 === 0,
+  }));
+
+  render(
+    <div>
+      <PageMenu variant="border" />
```

Comment:
You could wrap tests in `describe('border variant',` to distinguish between variants

Reply:
Done